### ISSUE TITLE
GoogleCloudEnterpriseSearchRetriever fails to where location is "us"

### ISSUE THREAD

**Issue by Adi8885** [langchain-ai/langchain#10580]
### System Info

'OS_NAME': 'DEBIAN_10'
Langchain version :  '0.0.288'
python :  3.10

### Information

- [ ] The official example notebooks/scripts
- [X] My own modified scripts

### Related Components

- [ ] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [X] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

This issue only occurs when region != "global". The retriever works well when region is set to "global"

Steps to reproduce :
1. Create a Enterprise search App with region = 'us'
![BHJRUwAReLjYUxb](https://github.com/langchain-ai/langchain/assets/31382824/89dc0e15-c8ce-4d64-be4c-9087705870da)

3. Import langchain version 0.0.288
```
import langchain
from langchain.retrievers import GoogleCloudEnterpriseSearchRetriever

PROJECT_ID = "<PROJECT_ID>"  # Set to your Project ID
SEARCH_ENGINE_ID = "<SEARCH_ENGINE_ID>"#"# Set to your data store ID

retriever = GoogleCloudEnterpriseSearchRetriever(
    project_id=PROJECT_ID,
    search_engine_id=SEARCH_ENGINE_ID,
    max_documents=3 ,
    location_id = "us"
)

retriever.get_relevant_documents("What is capital of India?")
```

4. Code errors out with below
```
---------------------------------------------------------------------------
_InactiveRpcError                         Traceback (most recent call last)
File /opt/conda/envs/python310/lib/python3.10/site-packages/google/api_core/grpc_helpers.py:72, in _wrap_unary_errors.<locals>.error_remapped_callable(*args, **kwargs)
     71 try:
---> 72     return callable_(*args, **kwargs)
     73 except grpc.RpcError as exc:

File /opt/conda/envs/python310/lib/python3.10/site-packages/grpc/_channel.py:1161, in _UnaryUnaryMultiCallable.__call__(self, request, timeout, metadata, credentials, wait_for_ready, compression)
   1155 (
   1156     state,
   1157     call,
   1158 ) = self._blocking(
   1159     request, timeout, metadata, credentials, wait_for_ready, compression
   1160 )
-> 1161 return _end_unary_response_blocking(state, call, False, None)

File /opt/conda/envs/python310/lib/python3.10/site-packages/grpc/_channel.py:1004, in _end_unary_response_blocking(state, call, with_call, deadline)
   1003 else:
-> 1004     raise _InactiveRpcError(state)

_InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
	status = StatusCode.NOT_FOUND
	details = "DataStore projects/PROJECT_ID/locations/us/collections/default_collection/dataStores/SEARCH_ENGINE_ID not found."
	debug_error_string = "UNKNOWN:Error received from peer ipv4:172.253.120.95:443 {created_time:"2023-09-14T12:55:00.327037809+00:00", grpc_status:5, grpc_message:"DataStore projects/PROJECT_ID/locations/us/collections/default_collection/dataStores/SEARCH_ENGINE_ID not found."}"
>

The above exception was the direct cause of the following exception:

NotFound                                  Traceback (most recent call last)
Cell In[365], line 1
----> 1 retriever.get_relevant_documents("What is capital of India?")

File ~/.local/lib/python3.10/site-packages/langchain/schema/retriever.py:208, in BaseRetriever.get_relevant_documents(self, query, callbacks, tags, metadata, **kwargs)
    206 except Exception as e:
    207     run_manager.on_retriever_error(e)
--> 208     raise e
    209 else:
    210     run_manager.on_retriever_end(
    211         result,
    212         **kwargs,
    213     )

File ~/.local/lib/python3.10/site-packages/langchain/schema/retriever.py:201, in BaseRetriever.get_relevant_documents(self, query, callbacks, tags, metadata, **kwargs)
    199 _kwargs = kwargs if self._expects_other_args else {}
    200 if self._new_arg_supported:
--> 201     result = self._get_relevant_documents(
    202         query, run_manager=run_manager, **_kwargs
    203     )
    204 else:
    205     result = self._get_relevant_documents(query, **_kwargs)

File ~/.local/lib/python3.10/site-packages/langchain/retrievers/google_cloud_enterprise_search.py:254, in GoogleCloudEnterpriseSearchRetriever._get_relevant_documents(self, query, run_manager)
    251 search_request = self._create_search_request(query)
    253 try:
--> 254     response = self._client.search(search_request)
    255 except InvalidArgument as e:
    256     raise type(e)(
    257         e.message + " This might be due to engine_data_type not set correctly."
    258     )

File /opt/conda/envs/python310/lib/python3.10/site-packages/google/cloud/discoveryengine_v1beta/services/search_service/client.py:577, in SearchServiceClient.search(self, request, retry, timeout, metadata)
    570 metadata = tuple(metadata) + (
    571     gapic_v1.routing_header.to_grpc_metadata(
    572         (("serving_config", request.serving_config),)
    573     ),
    574 )
    576 # Send the request.
--> 577 response = rpc(
    578     request,
    579     retry=retry,
    580     timeout=timeout,
    581     metadata=metadata,
    582 )
    584 # This method is paged; wrap the response in a pager, which provides
    585 # an `__iter__` convenience method.
    586 response = pagers.SearchPager(
    587     method=rpc,
    588     request=request,
    589     response=response,
    590     metadata=metadata,
    591 )

File /opt/conda/envs/python310/lib/python3.10/site-packages/google/api_core/gapic_v1/method.py:113, in _GapicCallable.__call__(self, timeout, retry, *args, **kwargs)
    110     metadata.extend(self._metadata)
    111     kwargs["metadata"] = metadata
--> 113 return wrapped_func(*args, **kwargs)

File /opt/conda/envs/python310/lib/python3.10/site-packages/google/api_core/grpc_helpers.py:74, in _wrap_unary_errors.<locals>.error_remapped_callable(*args, **kwargs)
     72     return callable_(*args, **kwargs)
     73 except grpc.RpcError as exc:
---> 74     raise exceptions.from_grpc_error(exc) from exc

NotFound: 404 DataStore projects/PROJECT_ID/locations/us/collections/default_collection/dataStores/SEARCH_ENGINE_ID not found.
```

### Expected behavior

Code should return three relevant documents from Enterprise Search

**Comment by holtskinner** [comment]
This was fixed in #10513 [`v0.0.310`](https://github.com/langchain-ai/langchain/releases/tag/v0.0.310)

Update to that version or greater and specify the location parameter.

Regional Data Stores require the endpoint to be set manually, which was not implemented in earlier versions of the retriever.

**Comment by Adi8885** [comment]
Thanks @holtskinner . I confirm that this works in higher versions.
